src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Yarp.ReverseProxy.Forwarder;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    internal class DefaultHttpProxyService : IHttpProxyService
+    {
+        private IHttpForwarder _httpForwarder;
+        private HttpMessageInvoker _messageInvoker;
+        private ForwarderRequestConfig _forwarderRequestConfig;
+        private string _proxyEndpoint;
+
+        public DefaultHttpProxyService(IHttpForwarder httpForwarder)
+        {
+            _httpForwarder = httpForwarder;
+
+            var handler = new SocketsHttpHandler();
+            _messageInvoker = new HttpMessageInvoker(handler);
+            _forwarderRequestConfig = new ForwarderRequestConfig();
+
+            // TODO: Update this logic. Port should come through configuration.
+            var port = Environment.GetEnvironmentVariable("AZURE_FUNCTIONS_HTTP_PROXY_PORT") ?? "5555";
+
+            _proxyEndpoint = "http://localhost:" + port;
+        }
+
+        public ValueTask<ForwarderError> Forward(ScriptInvocationContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.Inputs is null)
+            {
+                throw new ArgumentNullException(nameof(context.Inputs));
+            }
+
+            HttpRequest httpRequest = context.Inputs.FirstOrDefault(i => i.Val is HttpRequest).Val as HttpRequest;
+
+            if (httpRequest is null)
+            {
+                throw new InvalidOperationException($"Cannot proxy an HttpTrigger Function without an input of type {nameof(HttpRequest)}.");
+            }
+
+            HttpContext httpContext = httpRequest.HttpContext;
+
+            httpContext.Items.Add("IsHttpProxying", bool.TrueString);
+
+            // add invocation id as correlation id
+            // TODO: add "invocation-id" as a constant somewhere / maybe find a better name
+            httpRequest.Headers.TryAdd("invocation-id", context.ExecutionContext.InvocationId.ToString());
```

Comment:
Haven't gotten around to finalizing this name since I wasn't sure we were keeping this logic. At this point I think we are keeping it, so will take ur suggestion and try to finalize a name at the http sync.

Requires PR to change on worker side [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/befaead487e044a3e391587faf6d7f6397595ab4/extensions/Worker.Extensions.Http.AspNetCore/src/AspNetMiddleware.cs#L26) as well.

Reply:
A mix of headers use "X-" and "X-Functions" but we will revisit this before merging from feature branch to dev.